doc: documentation of flash_area and flash_map (#14841)
zephyrproject-rtos/zephyr: doc/reference/storage/flash_map/flash_mpa.rst

> .. _flash_map_flash_area:

Reviewer: Looks like the filename is misspelled, and should be ``flash_map.rst``

Flash map (NVS)
##########################

Flash map is a way for storing flash partitioning information in one central
distribution location in flash_area structures array form.

Flash map is generated from DTS based on content of :ref:`flash_partitions`
nodes.

> I order to use data contain in flash_map the flash_area API is provided.

Reviewer: ```suggestion
The flash_area API provides a way to access data in the flash map.
```

> flash_area_open() API is the interface for obtaining given flash partitions

Reviewer: ```suggestion
The flash_area_open() API is the interface for obtaining the flash partitions
```

flash_area from the flash map.

Flash Area API (flash_area)
###########################

> Flash_Area conception combines method for operation on flash chunk

Reviewer: ```suggestion
The flash_area concept combines methods for operating on a flash chunk
```

> with description of this chunk. Its methods are basically wrappers around

Reviewer: ```suggestion
together with a description of this chunk. Its methods are basically wrappers around
```

> flash API with addition of input parameters range check and so. Not all flash

Reviewer: ```suggestion
the flash API, with input parameter range checks. Not all flash
```

> operation are wrapped so API for retrieve flash area driver is included as well.

Reviewer: ```suggestion
operation are wrapped, so the API for retrieving data using the flash area driver is included as well.
```

Author: API for retrieve flash area driver - it really mean: to get the appropriate flash driver name which makes possible to operate on the area "by hand" using directly the flash driver API.

Author: @dbkinder ^^

> Flash area method are designed to be used along with flash_area structures of

Reviewer: ```suggestion
The flash area methods are designed to be used along with the flash_area structures of
```

> flash_map and users specific flash_areas with exception of area open API which

Reviewer: ```suggestion
flash_map and user-specific flash_areas, with the exception of the area_open API
```

> is used to fetch flash_area from flash_map.

Reviewer: ```suggestion
used to fetch a flash_area from the flash_map.
```
